### PR TITLE
Prepare REPL for dotty-bridge

### PR DESCRIPTION
**Pull request by felixmulder**
This PR is preparing the REPL for #1293 being completed

* [x] Add initial commands
* [x] Add cleanup commands
* [x] Add value binding functionality

### CODE REVIEW

File: src/dotty/tools/dotc/repl/CompilingInterpreter.scala [hunk 1]
```diff
@@ -228,6 +228,60 @@ class CompilingInterpreter(out: PrintWriter, ictx: Context) extends Compiler wit
...
+    setMethod.fold(false) { method =>
+      method.invoke(resObj, value) == null
+    }
+  }
+
+  /** This bind is implemented by creating an object with a set method and a
+   *  field `value`. The value is then set via Java reflection.
+   *
+   *  Example: We want to bind a value `List(1,2,3)` to identifier `list` from
+   *  SBT. The bind method accomplishes this by creating the following:
```
Comment (smarter) [inline src/dotty/tools/dotc/repl/CompilingInterpreter.scala]: It's called sbt, not SBT ;)

File: src/dotty/tools/dotc/repl/REPL.scala [hunk 2]
```diff
@@ -65,6 +65,9 @@ object REPL {
      */
     val cleanupCommands: List[String] = Nil
 
+    /** We also allow binding initial values */
```
Comment (smarter) [inline src/dotty/tools/dotc/repl/REPL.scala]: The description should not start with "We also allow ..." but just explains what bound values are and maybe give an example.

File: src/dotty/tools/dotc/repl/CompilingInterpreter.scala [hunk 3]
```diff
@@ -738,6 +797,10 @@ class CompilingInterpreter(out: PrintWriter, ictx: Context) extends Compiler wit
     INTERPRETER_LINE_PREFIX + num
   }
 
+  /** get the current line name */
```
Comment (smarter) [inline src/dotty/tools/dotc/repl/CompilingInterpreter.scala]: Capital letter at the start of a doc comment, but here the method is private and the comment has no information that's not in the method name, so I just wouldn't put a comment

**Comment by felixmulder** [comment]
@smarter: fixed your feedback 👍

File: src/dotty/tools/dotc/repl/CompilingInterpreter.scala [hunk 4]
```diff
@@ -228,6 +228,65 @@ class CompilingInterpreter(out: PrintWriter, ictx: Context) extends Compiler wit
...
+
+    try {
+      val resObj: Class[_] = Class.forName(objName, true, classLoader)
+      val setMethod = resObj.getDeclaredMethods.find(_.getName == "set")
+
+      setMethod.fold(false) { method =>
+        method.invoke(resObj, value) == null
+      }
+    } catch {
+      case _: Throwable =>
```
Comment (smarter) [inline src/dotty/tools/dotc/repl/CompilingInterpreter.scala]: `import scala.util.control.NonFatal` then `case NonFatal(_) =>` instead?

**Comment by smarter** [comment]
Otherwise LGTM

**Comment by smarter** [comment]
LGTM needs to be at the beginning of the line otherwise dotty-bot is unhappy :)

**Comment by DarkDimius** [comment]
LGTM

**Pull request merged**